src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -788,6 +766,88 @@ public SolutionCompilationState WithProjectAnalyzerReferences(
                     stateChange.OldProjectState, stateChange.NewProjectState, referencesToAdd: addedReferences, referencesToRemove: removedReferences);
             },
             forkTracker: true);
+
+        // As we're changing the references for a project we want to force generators to rerun for it.  This way the
+        // user automatically sees the new generators values without having to take any more explicit steps.
+        var updatedMap = GetUpdatedSourceGeneratorVersions(this,
+            ImmutableSegmentedList<(ProjectId? projectId, bool forceRegeneration)>.Empty.Add((projectId, true)));
```

Comment:
I'm assuming this is faster than just `[(projectId, true)]`?

Reply:
unfortunately, that type doesn't expose collection initializer support yet.  @sharwell  for segmented list support.